reactor-core/src/test/java/reactor/core/publisher/FluxBufferTimeoutFairBackpressureTest.java

```diff
@@ -365,7 +367,7 @@ public void requestedFromUpstreamShouldNotExceedDownstreamDemand() {
 		            .assertNext(s -> assertThat(s).containsExactly("a"))
 		            .then(() -> assertThat(requestedOutstanding).hasValue(19))
 		            .thenRequest(1)
-		            .then(() -> assertThat(requestedOutstanding).hasValue(20))
+		            .then(() -> assertThat(requestedOutstanding).hasValue(19))
```

Comment:
This is a behaviour change, yet it is not dramatic or disruptive. Before, a `request` would force `request`ing whatever is remaining from the upstream to fulfill the `prefetch` capacity. Now it will do that only once the `replenishMark` is reached.